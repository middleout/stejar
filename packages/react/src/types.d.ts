declare module "react-dom-factories" {
    export default (className: any): any => null;
}
