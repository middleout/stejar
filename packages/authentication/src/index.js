export { authenticated, logout } from "./actions";
export { AUTHENTICATED_ACTION, LOGOUT_ACTION } from "./actionTypes";
export { authenticationReducer } from "./reducer";
export { hasIdentity, getIdentity } from "./selectors";
export {
    getActionsPrefix,
    setActionsPrefix,
    getStateSelector,
    setStateSelector,
    setMountPoint,
    getMountPoint,
} from "./settings";
