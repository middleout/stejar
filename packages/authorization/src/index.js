export { addPermission, loadedPermissions, loadedResources, loadedRoles } from "./actions";
export {
    ADDED_PERMISSION_ACTION,
    LOADED_PERMISSIONS_ACTION,
    LOADED_RESOURCES_ACTION,
    LOADED_ROLES_ACTION,
} from "./actionTypes";
export { authorizationReducer } from "./reducer";
export { getResources, getRoles, getRules, isAllowed } from "./selectors";
export {
    getActionsPrefix,
    setActionsPrefix,
    getStateSelector,
    setStateSelector,
    setMountPoint,
    getMountPoint,
} from "./settings";
