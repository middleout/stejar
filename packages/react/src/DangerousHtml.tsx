import DOM from "react-dom-factories";

/**
 * @param props
 * @returns {any}
 * @constructor
 */
export function DangerousHtml(props: any) {
    const newProps: any = { ...props };
    (newProps as any).dangerouslySetInnerHTML = {
        __html: props.children as string,
    };
    delete newProps.children;
    const type = newProps.type;
    delete newProps.type;
    if (!(newProps as any).className) {
        (newProps as any).className = "dangerousHtml";
    } else {
        (newProps as any).className += " dangerousHtml";
    }
    return DOM[type || "span"](newProps);
}
