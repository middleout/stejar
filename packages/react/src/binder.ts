export function binder<T>(object: T): T {
    let keys: any[] = [];

    if (typeof Reflect !== "undefined" && typeof Reflect.ownKeys === "function") {
        keys = Reflect.ownKeys(object.constructor.prototype);
    } else {
        keys = Object.getOwnPropertyNames(object.constructor.prototype);
        if (typeof Object.getOwnPropertySymbols === "function") {
            keys = keys.concat(Object.getOwnPropertySymbols(object.constructor.prototype));
        }
    }

    keys.forEach(key => {
        // Ignore special case target method
        if (key === "constructor") {
            return;
        }

        if (typeof this[key] === "function") {
            this[key] = this[key].bind(this);
        }
    });

    return object;
}
