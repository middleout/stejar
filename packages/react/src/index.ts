export * from "./Component";
export * from "./PureComponent";
export * from "./DangerousHtml";
