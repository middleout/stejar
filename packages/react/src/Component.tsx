import { Component as ReactComponent } from "react";
import { binder } from "./binder";

export abstract class Component<P = {}, S = {}> extends ReactComponent<P, S> {
    props: P & { children?: any };

    /**
     * @param props
     * @param context
     */
    constructor(props: P, context?: any) {
        super(props, context);
        binder(this);
    }

    /**
     * @returns {null}
     */
    componentWillMount(): void {
        return null;
    }

    /**
     * This method is used just for auto completition in editors (override method)
     *
     * @returns {JSX.Element}
     */
    abstract render(): JSX.Element;

    /**
     * This method is used just for auto completition in editors (override method)
     *
     * @returns {null}
     */
    componentDidMount(): void {
        return null;
    }

    /**
     * This method is used just for auto completition in editors (override method)
     *
     * @param nextProps
     * @returns {null}
     */
    componentWillReceiveProps(nextProps: P): void {
        return null;
    }

    /**
     * This method is used just for auto completition in editors (override method)
     *
     * @returns {null}
     */
    shouldComponentUpdate(nextProps: P, nextState: S): boolean {
        return true;
    }

    /**
     * This method is used just for auto completition in editors (override method)
     *
     * @returns {null}
     */
    componentWillUpdate(nextProps: P, nextState: S): void {
        return null;
    }

    /**
     * This method is used just for auto completition in editors (override method)
     *
     * @returns {null}
     */
    componentDidUpdate(prevProps: P, prevState: S): void {
        return null;
    }

    /**
     * This method is used just for auto completition in editors (override method)
     *
     * @returns {null}
     */
    componentWillUnmount(): void {
        return null;
    }
}
