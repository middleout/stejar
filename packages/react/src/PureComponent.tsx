import { PureComponent as ReactPureComponent } from "react";
import { binder } from "./binder";

export abstract class PureComponent<P = {}, S = {}> extends ReactPureComponent<P, S> {
    props: P & { children?: any };

    /**
     * @param props
     * @param context
     */
    constructor(props: P, context?: any) {
        super(props, context);
        binder(this);
    }

    /**
     * This method is used just for auto completition in editors (override method)
     *
     * @returns {null}
     */
    componentWillMount(): void {
        return null;
    }

    /**
     * This method is used just for auto completition in editors (override method)
     *
     * @returns {JSX.Element}
     */
    abstract render(): JSX.Element;

    /**
     * This method is used just for auto completition in editors (override method)
     *
     * @returns {null}
     */
    componentDidMount(): void {
        return null;
    }

    /**
     * This method is used just for auto completition in editors (override method)
     *
     * @param nextProps
     * @returns {null}
     */
    componentWillReceiveProps(nextProps: P): void {
        return null;
    }

    /**
     * This method is used just for auto completition in editors (override method)
     *
     * @returns {null}
     */
    shouldComponentUpdate(nextProps: P, nextState: S): boolean {
        return true;
    }

    /**
     * This method is used just for auto completition in editors (override method)
     *
     * @returns {null}
     */
    componentWillUpdate(nextProps: P, nextState: S): void {
        return null;
    }

    /**
     * This method is used just for auto completition in editors (override method)
     *
     * @returns {null}
     */
    componentDidUpdate(prevProps: P, prevState: S): void {
        return null;
    }

    /**
     * This method is used just for auto completition in editors (override method)
     *
     * @returns {null}
     */
    componentWillUnmount(): void {
        return null;
    }
}
