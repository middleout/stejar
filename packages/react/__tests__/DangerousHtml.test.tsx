import { render } from "enzyme";
import React from "react";
import renderer from "react-test-renderer";
import { DangerousHtml } from "../src/DangerousHtml";

test("Should be able to use it", () => {
    const component = renderer.create(<DangerousHtml>Hello World</DangerousHtml>);

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    const dom = render(<DangerousHtml>Hello World</DangerousHtml>);
    expect(dom.text()).toEqual("Hello World");
});
