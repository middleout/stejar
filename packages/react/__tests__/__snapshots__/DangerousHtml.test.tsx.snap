// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should be able to use it 1`] = `
<span
  className="dangerousHtml"
  dangerouslySetInnerHTML={
    Object {
      "__html": "Hello World",
    }
  }
/>
`;
